### PR TITLE
feat: parse invariant TimeSpan durations

### LINKED ISSUE #691: [Feature Request] TimeSpan.Dehumanize()
Yes, I know there's been some lengthy discussion about dehumanizing dates (#112). And the [readme](https://github.com/Humanizr/Humanizer/blob/dev/readme.md) even states: 

> No dehumanization for dates as `Humanize` is a lossy transformation and the human friendly date is not reversible"

There's also already a couple of libraries that deal with ambiguous expressions such as "yesterday", "3 weeks ago", and "7 hours before tomorrow at midnight". e.g. [nChronic](https://github.com/robertwilczynski/nChronic/) and Microsoft LUIS.

However, I can't see any discussion about dehumanizing `TimeSpan`s, which are much more straight forward. Even if the main focus here is on humanizing rather than the reverse, this kind of thing would be well suited to this library

## TimeSpan

`TimeSpan`s are much less ambiguous than dates. There also don't seem to be many other libraries that deal with them well. Here's some basic examples / semi-pseudocode tests for a TimeSpan dehumanizer: 

```
var timeTest = new TimeSpan(3, 18, 0)
Assert.Equal(TimeSpan.Dehumanize("three hours and eighteen minutes"), timeTest);
Assert.Equal(TimeSpan.Dehumanize("3 hours, 18 minutes"), timeTest);
Assert.Equal(TimeSpan.Dehumanize("3:18 hours"), timeTest);
Assert.Equal(TimeSpan.Dehumanize("3h18m"), timeTest);
Assert.Equal(TimeSpan.Dehumanize("3h 18m"), timeTest);
Assert.Equal(TimeSpan.Dehumanize("3.3hrs"), timeTest);
Assert.Equal(TimeSpan.Dehumanize("3:18:00"), timeTest);
Assert.Equal(TimeSpan.Dehumanize("3:18"), timeTest);
Assert.Equal(TimeSpan.Dehumanize("18:01", expectedUnits:minutes), new TimeSpan(0, 18, 01);
```

I think this kind of thing would be fairly straight forward and without ambiguity.

The other libraries I've found that deal with dehumanizing TimeSpans are: 
* Microsoft LUIS, e.g. [`builtin.datetimeV2.duration`](https://docs.microsoft.com/en-us/azure/cognitive-services/luis/luis-reference-prebuilt-entities#builtindatetimev2) but it seems like it's overly complex and I'm guessing it's not open source. Also seems more focused on speech than text.
* [Exceptionless.DateTimeExtensions](https://github.com/exceptionless/Exceptionless.DateTimeExtensions#timeunit) (Apache 2.0 licensed) has a parser that works as following: `TimeSpan oneHour = TimeUnit.Parse("3h");` But would fail on all of the example strings I've given above as they are "complex".

TL;DR: Humanizer would be a good library to add a TimeSpan Dehumanizer to. I could be wrong, but I can't see anything overly complicated about it (unlike dehumanizing TimeDate text), as long as you stay away from accepting highly ambiguous inputs like "2 months" and stick to making it work as would be expected. I don't know much about what sort of things might trip up localization efforts though, but making it part of Humanizer would give a good chance for localization to occur.

I haven't (yet) poked around with the inner workers of Humanizer's parsers to see how easy it might be to add something like this, but thought I'd put it out there to see what interest was like.

### PR DESCRIPTION
**Pull request by clairernovotny**
## Summary

Applications can now parse standard invariant `TimeSpan` text and compact durations such as `3h18m` or `1.5w 2d` through throwing and `Try` APIs. Standard colon forms retain the BCL's invariant interpretation; compact parsing accepts only `ms`, `s`, `m`, `h`, `d`, and `w`, applies one leading sign to the whole duration, and uses exact tick arithmetic at overflow boundaries.

This first slice deliberately excludes alternate colon interpretations, localized phrases, number words, months, and years. Thanks to @pengowray for the original report and use cases, and to @wyf027 for the implementation exploration in #1800 and #1801 that informed the compatible API names and edge-case coverage.

## Validation

- Focused `TimeSpanDehumanizeTests` on net10.0: 44 passed
- Full `Humanizer.Tests` suite on net8.0, net10.0, and net11.0: 57,717 passed on each target
- Release pack across netstandard2.0, net48, net8.0, net10.0, and net11.0: succeeded without warnings
- `dotnet format Humanizer.slnx --verify-no-changes --verbosity minimal --no-restore`: passed
- Independent 25,000-case exact-arithmetic fuzz comparison: no mismatches
- Linux cannot execute net48 tests; Windows CI remains authoritative for net48 compilation and test execution

## Checklist

- [x] Implementation is clean
- [x] Code adheres to the existing coding standards
- [x] No Code Analysis warnings
- [x] There is proper unit test coverage
- [x] No code was copied from StackOverflow, a blog, or another OSS implementation
- [x] Comments are limited to XML documentation
- [x] XML documentation is included for the new public APIs
- [x] Rebased on the latest `main`
- [x] README documents the supported contract
- [x] Repository tests and release packing pass

Fixes #691

### CODE REVIEW

File: src/Humanizer/TimeSpanDehumanizeExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+            if (index < input.Length && input[index] == '.')
+            {
+                index++;
+                fractionStart = index;
+                while (index < input.Length && IsAsciiDigit(input[index]))
+                {
+                    index++;
+                    digitCount++;
+                }
+            }
```
[the remark this answers is not in the document]
Comment (clairernovotny) [reply src/Humanizer/TimeSpanDehumanizeExtensions.cs:99]: > Compact parsing currently accepts a decimal point with no fractional digits.

Fixed in 5ef04c88 — compact tokens now require at least one digit after `.`, with regression coverage for `1.h` and `1.ms`.

**Comment by clairernovotny** [comment]
> The new parser mixes brace styles for single-statement `if` blocks.

Fixed in ec693647 — removed unnecessary braces from all single-statement `if` bodies while retaining braces for multi-statement bodies. This is style-only.
